cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1617,8 +1617,17 @@ func (p *BicepProvider) loadParameters(ctx context.Context) (loadParametersResul
 			resolvedParams[paramName] = resolvedParam
 			continue
 		}
+
+		// Check if the parameter has a valid type
 		stringValue, isString := resolvedParam.Value.(string)
-		if !isString {
+		mapValue, isMap := resolvedParam.Value.(map[string]any)
+		_, isArray := resolvedParam.Value.([]any)
+		if !isString && !isMap && !isArray {
```

Comment:
thank you for your suggestion @Mstiekema.  It is a good one, but we need something different.

The intention of checking if the parameter's value type is string is to handle a very unique case for string parameters where we don't know if the value is empty b/c it s mapped to an ENV VAR that is not defined or b/c it is mapped to an ENV VAR that is defined with an empty string.

We are trying just to ignore the first case when we were setting an empty string after a missing ENV VAR.

I made a mistake by ignoring any non-string parameter here. We only want to ignore string parameters with empty value, non-nil and hasUnsetEnvVar = true.